cdap-docs/admin-manual/source/operations/logging.rst

```diff
@@ -50,28 +102,409 @@ CDAP supports logging through standard `SLF4J (Simple Logging Facade for Java)
+    ``<cdap-sdk-home>/conf``, configures both "container" and CDAP system service logging.
+
+
+Configuring the Log Saver Service
+=================================
+The Log Saver Service is the CDAP service that reads log messages from Kafka, processes
+them in *log pipelines*, persists them to HDFS, and sends metrics on logging to the
+Metrics Service.
+
+Properties and configurations that control the Log Saver Service are located in the
+:ref:`cdap-site.xml <appendix-cdap-site.xml>` file.
+
+CDAP Site XML File
+------------------
+The :ref:`cdap-site.xml <appendix-cdap-site.xml>` file has properties that control the
+writing of logs to Kafka, the CDAP Log Saver Service, the CDAP log pipeline, and any
+:ref:`custom log pipelines <logging-monitoring-custom-logging-pipeline>` that have been defined.
+
+Writing Logs to Kafka
+.....................
+These :ref:`properties <appendix-cdap-default-logging>` control the writing of logs to
+Kafka:
+
+.. generated by code in conf.py for this manual
+.. include:: /../target/_includes/logging-writing-logs-to-kafka.rst
+
+Log Saver Service
+.................
+These :ref:`properties <appendix-cdap-default-logging>` control the Log Saver Service:
+
+.. generated by code in conf.py for this manual
+.. include:: /../target/_includes/logging-log-saver-service.rst
 
-- Using the :ref:`restful-api`. The :ref:`Logging HTTP RESTful API <http-restful-api-logging>` 
-  details the available contexts that can be called to retrieve different messages.
-- Log messages of an application can be viewed in the :ref:`CDAP UI <cdap-ui>`
-  by clicking the *Logs* tab of an application.
+Log saver instances should be from a minimum of one to a maximum of ten. The maximum is
+set by the number of Kafka partitions (``log.publish.num.partitions``), which by default is 10.
 
+.. _logging-monitoring-logging-pipeline-configuration:
 
-.. _master-service-logging-configuration:
+Logging Pipeline Configuration
+..............................
+Configuration properties for logging and :ref:`custom logging pipelines
+<logging-monitoring-custom-logging-pipeline>` are shown in the documentation of the
+:ref:`logging properties <appendix-cdap-default-logging>` section of the
+:ref:`cdap-site.xml <appendix-cdap-site.xml>` file.
 
-Master Service Logging Configuration
-====================================
+The **CDAP log pipeline** is configured by settings in the :ref:`cdap-site.xml
+<appendix-cdap-site.xml>` file.
 
-:term:`Master Services <master services>` use ``logback-container.xml``, packaged with the CDAP distribution,
-for logging configuration. This logback does log rotation once a day at midnight and expires logs older than
-14 days. Changes can be made to ``logback-container.xml``, and the ``cdap-master`` process will need to be restarted
-for the modified logback to take effect.
+**Custom logging pipelines** are configured by a combination of the settings in the 
```

Comment:
Did you introduce/motivate the term "custom log pipeline" previously? It seems to fall out of the sky here.